src/CalcViewModel/DataLoaders/CurrencyDataLoader.cpp

```diff
@@ -31,41 +33,84 @@ using namespace Windows::System::UserProfile;
-
-static constexpr auto ALL_RATIOS_DATA_FILENAME = L"CURRENCY_CONVERTER_ALL_RATIOS_DATA.txt";
-static constexpr auto RATIO_KEY = L"Rt";
-static constexpr auto CURRENCY_CODE_KEY = L"An";
-static constexpr array<wstring_view, 2> ALL_RATIOS_DATA_PROPERTIES = { wstring_view{ RATIO_KEY, 2 }, wstring_view{ CURRENCY_CODE_KEY, 2 } };
-
-static constexpr auto DEFAULT_FROM_TO_CURRENCY_FILE_URI = L"ms-appx:///DataLoaders/DefaultFromToCurrency.json";
-static constexpr auto FROM_KEY = L"from";
-static constexpr auto TO_KEY = L"to";
+namespace
+{
+    constexpr auto CURRENCY_UNIT_FROM_KEY = L"CURRENCY_UNIT_FROM_KEY";
+    constexpr auto CURRENCY_UNIT_TO_KEY = L"CURRENCY_UNIT_TO_KEY";
+
+    // Calculate number of 100-nanosecond intervals-per-day
+    // (1 interval/100 nanosecond)(100 nanosecond/1e-7 s)(60 s/1 min)(60 min/1 hr)(24 hr/1 day) = (interval/day)
+    constexpr long long DAY_DURATION = 1LL * 60 * 60 * 24 * 10000000;
+    constexpr long long WEEK_DURATION = DAY_DURATION * 7;
+
+    constexpr int FORMATTER_RATE_FRACTION_PADDING = 2;
+    constexpr int FORMATTER_RATE_MIN_DECIMALS = 4;
+    constexpr int FORMATTER_RATE_MIN_SIGNIFICANT_DECIMALS = 4;
+
+    constexpr auto CACHE_TIMESTAMP_KEY = L"CURRENCY_CONVERTER_TIMESTAMP";
+    constexpr auto CACHE_LANGCODE_KEY = L"CURRENCY_CONVERTER_LANGCODE";
+    constexpr auto CACHE_DELIMITER = L"%";
+
+    constexpr auto STATIC_DATA_FILENAME = L"CURRENCY_CONVERTER_STATIC_DATA.txt";
+    constexpr array<wstring_view, 5> STATIC_DATA_PROPERTIES = { wstring_view{ L"CountryCode", 11 },
+                                                                wstring_view{ L"CountryName", 11 },
+                                                                wstring_view{ L"CurrencyCode", 12 },
+                                                                wstring_view{ L"CurrencyName", 12 },
+                                                                wstring_view{ L"CurrencySymbol", 14 } };
+
+    constexpr auto ALL_RATIOS_DATA_FILENAME = L"CURRENCY_CONVERTER_ALL_RATIOS_DATA.txt";
+    constexpr auto RATIO_KEY = L"Rt";
+    constexpr auto CURRENCY_CODE_KEY = L"An";
+    constexpr array<wstring_view, 2> ALL_RATIOS_DATA_PROPERTIES = { wstring_view{ RATIO_KEY, 2 }, wstring_view{ CURRENCY_CODE_KEY, 2 } };
+
+    constexpr auto DEFAULT_FROM_TO_CURRENCY_FILE_URI = L"ms-appx:///DataLoaders/DefaultFromToCurrency.json";
+    constexpr auto FROM_KEY = L"from";
+    constexpr auto TO_KEY = L"to";
+
+    // Fallback default values.
+    constexpr auto DEFAULT_FROM_CURRENCY = DefaultCurrencyCode.data();
+    constexpr auto DEFAULT_TO_CURRENCY = L"EUR";
+
+    // ParseLanguageCode returns language code in form of `lang-region`
+    // TODO: unit testing.
+    std::optional<std::wstring> ParseLanguageCode(const wchar_t* bcp47)
+    {
+        // the IETF BCP 47 language tag syntax is: language[-script][-region][-variant[*][-extension*][-privateuse*]]
```

Comment:
nit: extra `[`?

Reply:
removed the trailing parts that this parser doesn't care.